TESTS/API/I2C/I2C.cpp

```diff
@@ -58,7 +58,7 @@ void test_lm75b()
 template<PinName sda, PinName scl,int size_of_data, int address>
 void flash_WR()
 {
-    I2CEeprom memory(sda,scl,MBED_CONF_APP_I2C_EEPROM_ADDR,32,0);
+    I2CEEBlockDevice memory(sda,scl,MBED_CONF_APP_I2C_EEPROM_ADDR,32,0);

```

Comment:
32 was pageSize, see
`I2CEeprom(PinName sda, PinName scl, int address, size_t pageSize, size_t chipSize, int busSpeed = 400000);`

so we should keep the default value of 32 in `I2CEEBlockDevice(
            PinName sda, PinName scl, uint8_t address,
            bd_size_t size, bd_size_t block=32,
            int bus_speed=400000);`

And `bd_size_t size` should be set to cover the test range, so at least `address` + `size_of_data` (1100 bytes). You could use
```
#define TEST_BLOCK_SIZE    32
#define TEST_SIZE                  TEST_BLOCK_SIZE * 500

```